sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/CognitiveComplexity.cs

```diff
@@ -759,5 +759,25 @@ void BlockAction()
                     }
                 };
         }
+
+        void NullCoalescence(string first, string second) // Compliant - FN: null-coalescing operator is not considered: https://github.com/SonarSource/sonar-dotnet/issues/3528
+        {
+            var x = first ?? second;
+        }
+
+        void NullCoalescenceAssignment(int? first, int? second, int? third, int? fourth) // Noncompliant {{Refactor this method to reduce its Cognitive Complexity from 8 to the 0 allowed.}}
+        {
+            var a = first ??= 0;
+//                        ^^^ Secondary {{+1}}
+
+            var b = first ??= second ??= third ??= fourth;
```

Comment:
Can you add a test demonstrating broken chain? Something like `a && b || c && d`? In this case, probably `fist ??= second = third ??= fourth`.